### PR TITLE
[Integration Test] Stream negotiation and fallback

### PR DESCRIPTION
**Pull request by yhassanzadeh13**
This PR fixes a bug with stream negotiation and fallback, and also adds integration testing for stream compression between execution and verification nodes. It also extends unit testing of libp2p streams with more happy and unhappy path scenarios.

### CODE REVIEW

File: integration/tests/verification/verification_test.go [hunk 1]
```diff
@@ -22,17 +26,7 @@ type VerificationTestSuite struct {
...
-	blockA := suite.BlockState.WaitForHighestFinalizedProgress(suite.T())
-	suite.T().Logf("blockA generated, height: %v ID: %v\n", blockA.Header.Height, blockA.Header.ID())
-
-	// waits for execution receipt for blockA from execution node, called receiptA
-	receiptA := suite.ReceiptState.WaitForReceiptFrom(suite.T(), blockA.Header.ID(), suite.exe1ID)
-	resultID := receiptA.ExecutionResult.ID()
-	suite.T().Logf("receipt for blockA generated: result ID: %x\n", resultID)
-
-	// wait for a result approval from verification node
-	suite.ApprovalState.WaitForResultApproval(suite.T(), suite.verID, resultID, uint64(0))
```
Comment (yhassanzadeh13) [inline integration/tests/verification/verification_test.go:35]: Just moves it into a separate sub-function that shares between methods 👇

File: network/p2p/libp2pNode.go [hunk 2]
```diff
@@ -177,7 +177,10 @@ func (n *Node) CreateStream(ctx context.Context, peerID peer.ID) (libp2pnet.Stre
 			dialAddrs, err))
 	}
 
-	lg.Debug().Str("dial_address", fmt.Sprintf("%v", dialAddrs)).Msg("stream successfully created to remote peer")
+	lg.Info().
+		Str("networking_protocol_id", string(stream.Protocol())).
+		Str("dial_address", fmt.Sprintf("%v", dialAddrs)).
+		Msg("stream successfully created to remote peer")
```
Comment (yhassanzadeh13) [inline network/p2p/libp2pNode.go:183]: Bumping up log level to provide insight on the established protocol between nodes while negotiation. This helps on responding to incidents.

**Review by jwinkler2083233 — APPROVED** [review]
Looks good.
I'd like to see much more configurability in future releases of this, like the compression level.

File: integration/tests/verification/streamNegotiation_test.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+)
+
+// TestVerificationStreamNegotiationSuite enables gzip stream compression only between execution and verification nodes, while the
+// rest of network runs on plain libp2p streams. It evaluates that network operates on its happy path concerning verification functionality.
+func TestVerificationStreamNegotiationSuite(t *testing.T) {
+	s := new(VerificationStreamNegotiationSuite)
+	s.preferredUnicasts = "gzip-compression" // enables gzip stream compression between execution and verification node
```
Comment (vishalchangrani) [inline integration/tests/verification/streamNegotiation_test.go:13]: shouldn't `gzip-compression` be a constant defined in the main code?

Comment (yhassanzadeh13) [reply integration/tests/verification/streamNegotiation_test.go]: Addressed in: https://github.com/onflow/flow-go/pull/1812/commits/60acaadd40153ddcb231befaa72e25481389e140

**Review by vishalchangrani — APPROVED** [review]

**Review by jwinkler2083233 — COMMENTED** [review]
Looks good.

**Pull request merged**